firebase-database/src/main/java/com/google/firebase/database/core/utilities/DefaultRunLoop.java

```diff
@@ -128,13 +128,19 @@ public static String messageForException(Throwable t) {
           + "https://firebase.google.com/docs/database/ios/structure-data#best_practices_for_data_structure"
           + " and "
           + "https://firebase.google.com/docs/database/android/retrieve-data#filtering_data";
+    } else if (t instanceof NoClassDefFoundError) {
+      return "A symbol that the Firebase Database SDK depends on failed to load. This usually "
+          + "indicates that your project includes an incompatible version of another Firebase "
+          + "dependency. If updating your dependencies to the latest version does not resolve "
+          + "this issue, please file a report at https://github.com/firebase/firebase-android-sdk";
```

Comment:
If you want to be more helpful, change these URLs to https://github.com/firebase/firebase-android-sdk/issues/new?template=bug.md

If you want to be less helpful, change it to "on github" 😛

Reply:
Left it as is for now. I deem it to be the right level of helpfulness :)